test/index-test.js

```diff
@@ -6,13 +6,106 @@ const path = require('path')
 
 
 describe('index', () => {
-  
+
   jsdom({
+    html: fs.readFileSync(path.resolve(__dirname, '..', 'index.html'), 'utf-8'),
     src: fs.readFileSync(path.resolve(__dirname, '..', 'index.js'), 'utf-8')
   })
-  
 
-  it('runs', () => {
-    expect(true).to.be.true
+  describe('page', () => {
+    before(() => {
+      window.Handlebars = require('handlebars')
+      init()
+    })
+
+    it('has a recipe form', () => {
+      var recipeForm = document.getElementById("recipe-form")
+      expect(recipeForm).toExist("Must provide a form with an id of 'recipe-form'")
+      expect(recipeForm).toMatch(/onsubmit="createRecipe()"/)
+      var ingredients = document.getElementsByName("ingredients")
+      expect(ingredients.length).toBeGreaterThanOrEqualTo(5)
+      var nameField = document.getElementById("name")
+      expect(nameField).toExist()
+    })
+  })
+
+  describe('templates', () => {
+    it('has a recipe template', () => {
+      var recipeTemplate = document.getElementById("recipe-template")
+      expect(recipeTemplate).toExist("Must provide a template with an id of 'recipe-template'")
+      expect(recipeTemplate.type).toBe("text/x-handlebars-template", "Template must be of type text/x-handlebars-template")
+      expect(recipeTemplate.innerHTML).toMatch(/{{\s?name\s?}}/)
+      expect(recipeTemplate.innerHTML).toMatch(/<a.*displayEditForm().*>Edit Recipe<\/a>/, "Template must have an 'Edit Recipe' link that calls 'displayEditForm()'")
+      expect(recipeTemplate.innerHTML).toMatch(/{{>\s?recipeDetailsPartial\s?}}/, "Template must render the recipeDetailsPartial")
+    })
+    it('has a recipe details partial template', () => {
+      var recipeDetailsPartial = document.getElementById("recipe-details-partial")
+      expect(recipeDetailsPartial).toExist("Must provide a template with an id of 'recipe-details-partial'")
+      expect(recipeDetailsPartial.type).toBe("text/x-handlebars-template", "Template must be of type text/x-handlebars-template")
+      expect(recipeDetailsPartial.innerHTML).toMatch(/{{\s?description\s?}}/)
```

Comment:
yeah it goes against my normal testing principles but seemed like a good compromise to make sure at least they're doing the basic template stuff. took a while to land on something i liked for testing this one.